test-frame-common/src/main/java/io/skodjob/testframe/clients/KubeClient.java

```diff
@@ -98,6 +105,185 @@ public List<HasMetadata> readResourcesFromFile(InputStream is) throws IOExceptio
+                                "Creating", h.getKind(), h.getMetadata().getName());
+                        client.resource(h).inNamespace(namespace).create();
+                    } else {
+                        LOGGER.debug(LoggerUtils.RESOURCE_LOGGER_PATTERN,
+                                "Updating", h.getKind(), h.getMetadata().getName());
+                        client.resource(h).inNamespace(namespace).update();
+                    }
+                }
+            });
+        }
+    }
+
+    /**
+     * Create resources from file and apply modifier
+     * @param is stream of data
+     * @param modifier modifier method
+     * @throws IOException exception
+     */
+    public void create(InputStream is, Function<HasMetadata, HasMetadata> modifier) throws IOException {
+        try (is) {
+            client.load(is).get().forEach(i -> {
+                HasMetadata h = modifier.apply(i);
+                if (h != null) {
+                    if (client.resource(h).get() == null) {
+                        LOGGER.debug(LoggerUtils.RESOURCE_WITH_NAMESPACE_LOGGER_PATTERN,
+                                "Creating", h.getKind(), h.getMetadata().getName(), h.getMetadata().getNamespace());
+                        client.resource(h).create();
+                    } else {
+                        LOGGER.debug(LoggerUtils.RESOURCE_WITH_NAMESPACE_LOGGER_PATTERN,
+                                "Updating", h.getKind(), h.getMetadata().getName(), h.getMetadata().getNamespace());
+                        client.resource(h).update();
+                    }
+                }
+            });
+        }
+    }
+
+    /**
+     * Create resources from file and apply modifier
+     * @param namespace dest namesapce
+     * @param resources loaded resources
+     * @param modifier modifier method
+     */
+    public void create(String namespace, List<HasMetadata> resources, Function<HasMetadata, HasMetadata> modifier) {
+        resources.forEach(i -> {
+            HasMetadata h = modifier.apply(i);
+            if (h != null) {
+                if (client.resource(h).inNamespace(namespace).get() == null) {
+                    LOGGER.debug(LoggerUtils.RESOURCE_WITH_NAMESPACE_LOGGER_PATTERN,
+                            "Creating", h.getKind(), h.getMetadata().getName(), namespace);
+                    client.resource(h).inNamespace(namespace).create();
+                } else {
+                    LOGGER.debug(LoggerUtils.RESOURCE_WITH_NAMESPACE_LOGGER_PATTERN,
+                            "Updating", h.getKind(), h.getMetadata().getName(), namespace);
+                    client.resource(h).inNamespace(namespace).update();
+                }
+            }
+        });
+    }
+
+    /**
+     * Create resources from file and apply modifier
+     * @param resources resources
+     * @param modifier modifier method
+     */
+    public void create(List<HasMetadata> resources, Function<HasMetadata, HasMetadata> modifier) {
```

Comment:
I think that all the comments above are connected with one thing.
I would have one method that does the modification (using `modifier`) and the creation/update.

Others can then just use this method and either load it from the input stream, or use it for each of the resources in the list, or?

Reply:
well if you know how to use lambdas you know that you can skip modification with proper function

Reply:
Sorry, now I don't understand :D 

I meant to have one method for this:

```
                HasMetadata h = modifier.apply(i);
                if (h != null) {
                    if (client.resource(h).inNamespace(namespace).get() == null) {
                        LOGGER.debug(LoggerUtils.RESOURCE_LOGGER_PATTERN,
                                "Creating", h.getKind(), h.getMetadata().getName());
                        client.resource(h).inNamespace(namespace).create();
                    } 
                }
```
and then just use this method everywhere else.